### PR TITLE
shopinvader_response: can fail gracefully

### PR DESCRIPTION
**Pull request by simahawk**

### CODE REVIEW

File: shopinvader/shopinvader_response.py [hunk 1]
```diff
@@ -67,6 +67,14 @@ def get():
         # our context local data. (in a process/thread safe way). Since we are
         # in test mode we can rely on the threadLocal context
         current_local = threadLocal
-    if not hasattr(current_local, "_shopinvader_response"):
-        setattr(current_local, "_shopinvader_response", ShopinvaderResponse())
+    try:
+        if not hasattr(current_local, "_shopinvader_response"):
+            setattr(
```
Comment (lmignon) [inline shopinvader/shopinvader_response.py:72]: @simahawk Can you enlighten me as to when an error can occur with this kind of code?

Comment (simahawk) [reply shopinvader/shopinvader_response.py:72]: when this code is called outside of a shopinvader request you get this:
```
2020-01-17 13:50:44,950 1 ERROR odoodb odoo.addons.sale_automatic_workflow.models.automatic_workflow_job: Error during an automatic workflow action. 
Traceback (most recent call last):
  File "/odoo/external-src/sale-workflow/sale_automatic_workflow/models/automatic_workflow_job.py", line 23, in savepoint
    yield
  File "/odoo/external-src/sale-workflow/sale_automatic_workflow/models/automatic_workflow_job.py", line 94, in _sale_done
    sale.action_done()
  File "/odoo/src/addons/sale/models/sale.py", line 720, in action_done
    tx._set_transaction_done()
  File "/odoo/src/addons/payment/models/payment_acquirer.py", line 783, in _set_transaction_done
    'state_message': '',
  File "/odoo/external-src/odoo-shopinvader-payment/invader_payment/models/payment_transaction.py", line 45, in write
    self._notify_state_changed_event()
  File "/odoo/external-src/odoo-shopinvader-payment/invader_payment/models/payment_transaction.py", line 60, in _notify_state_changed_event
    payable._event(event_name).notify(payable, record)
  File "/odoo/external-src/connector/component_event/components/event.py", line 191, in notify
    event(*args, **kwargs)
  File "/odoo/external-src/odoo-shopinvader-payment/shopinvader_payment/components/payment_transaction_event_listerner.py", line 37, in on_payment_transaction_done
    def on_payment_transaction_done(self, sale_order, transaction):
  File "/odoo/external-src/odoo-shopinvader-payment/shopinvader_payment/components/payment_transaction_event_listerner.py", line 18, in _confirm_and_invalidate_session
    response = shopinvader_response.get()
  File "/odoo/external-src/odoo-shopinvader/shopinvader/shopinvader_response.py", line 70, in get
    if not hasattr(current_local, "_shopinvader_response"):
  File "/usr/local/lib/python3.7/dist-packages/werkzeug/local.py", line 348, in __getattr__
    return getattr(self._get_current_object(), name)
  File "/usr/local/lib/python3.7/dist-packages/werkzeug/local.py", line 307, in _get_current_object
    return self.__local()
  File "/usr/local/lib/python3.7/dist-packages/werkzeug/local.py", line 137, in _lookup
    raise RuntimeError("object unbound")
RuntimeError: object unbound
```

**Review by lmignon — APPROVED** [review]
Good catch @simahawk

**Pull request merged**